test/compatibility/test_fast_truncate_info.py

```diff
@@ -0,0 +1,174 @@
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import compatibility_test, os, shutil, sys
+
+class test_fast_truncate_info(compatibility_test.CompatibilityTestCase):
+    '''
+    Test handling of fast-truncate information across database upgrade and downgrade.
+    '''
+    cmake_args = '-DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_v4_clang.cmake -DENABLE_PYTHON=1'
+
+    # Use a small page size and lots of keys because we want to create lots
+    # of individual pages in the file.
+    config = 'allocation_size=512,leaf_page_max=512'
+    nentries = 10000
+    uri = 'file:test_fast_truncate'
+
+    def timestamp_str(self, t):
+        return '%x' % t
+
+    def test_fast_truncate_data_format(self):
+        '''
+        Test downgrading after performing a non globally-visible fast-truncate operation
+        on a major release newer than 6.1, which writes fast-truncate related information
+        into the database file. The older version should still be able to parse through
+        this fast-truncate information although it has no support to write it.
+        '''
+
+        # Checkout the relevant branches.
+        older_branch = 'mongodb-6.0'
+        newer_branch = 'mongodb-7.0'
+        self.prepare_wt_branch(older_branch, self.cmake_args)
+        self.prepare_wt_branch(newer_branch, self.cmake_args)
+
+        # Run the first part of the test on the 6.0 branch. This generates database files
+        # that do not contain fast-truncate information.
+        self.run_method_on_branch(older_branch, 'generate_older_database_files')
+
+        # Run the second part of the test on the 7.0 branch. This reads 6.0 database files
+        # and appends fast-truncate information onto database records when needed.
+        self.run_method_on_branch(newer_branch, 'write_fast_truncate_info')
+
+        # Run the third part of the test on the 6.0 branch. This attempts to read database
+        # files containing fast-truncate information from an older version that does not
+        # support writing out fast-truncate information. We should be able to read the
+        # information though we will simply ignore it.
+        self.run_method_on_branch(older_branch, 'read_fast_truncate_info_from_older')
+
+    def generate_older_database_files(self):
+        import wiredtiger
+
+        conn = wiredtiger.wiredtiger_open('.', 'create,')
+        session = conn.open_session()
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+        session.create(self.uri, self.config + ',key_format=i,value_format=S')
+
+        c = session.open_cursor(self.uri)
+        for i in range(1, self.nentries):
+            c[i] = 'i' + str(i)
+        c.close()
+
+        session.checkpoint()
+
+        session.close()
+        conn.close
```

Comment:
Did you mean to say `conn.close()`? You are missing the last `()`, so as it is now, the method would not be invoked.